ir/src/integrity_stmts/mod.rs

```diff
@@ -123,3 +126,28 @@ impl IntegrityStmts {
         Ok(())
     }
 }
+
+/// A [ConstraintRoot] represents the entry node of a subgraph representing an integrity constraint
+/// within the [AlgebraicGraph]. It also contains the row offsets for the constraint.
```

Comment:
I think we need to define what is meant by "the row offsets for the constraint"

edit: also, as noted in my other comments,  I think this should change to "the row offset"

Reply:
> I think we need to define what is meant by "the row offsets for the constraint"

I still think this could be addressed better. I left a new comment inline